www.ocaml.org.yml

```diff
@@ -93,6 +95,11 @@
           }
           
           sub vcl_recv {
+              if (vsthrottle.is_denied(client.identity, 15, 10s, 30s)) {
+                  # Client has exceeded 15 reqs per 10s.
```

Comment:
We'd need to test what a reasonable rate is.  According to Chrome network manager, just opening ocaml.org creates 45 requests.  How are requests counted?

Reply:
Great catch here. Thanks for that, and sorry for the delay in addressing it.

I fixed this in https://github.com/mtelvers/ansible/pull/37/commits/f8c57c691d0ef9cd4c8a1abacc0cbf5bfee3ae80 with an implausibly high limit of requests per second, which would ensure no legit users have problems.

However, requests are counted based in IP when using `client.identity`. If multiple users are trying to access ocaml.org from the same IP (I'm imagining students in a large lecture all going thru the same WiFi router), we may rate limit the bunch. There are methods to try to differentiate distinct users sharing the same IP address (see, e.g., https://blog.cloudflare.com/multi-user-ip-address-detection/).

But now I am thinking that maybe we should not proactively try to solve a problem that we have not faced, and which may be better addressed thru using Scaleway's https://www.scaleway.com/en/docs/dedibox-network/network/reference-content/anti-ddos/ ...

WDYT? Should we just close this?

Reply:
This closed automatically when I merged the other PR.  However, I do thnk we should park this until it becomes an issue.

Reply:
Yes, good idea.  We can keep sight of this potential fix.